clients/da-vinci-client/src/main/java/com/linkedin/davinci/config/VeniceServerConfig.java

```diff
@@ -477,6 +489,8 @@ public VeniceServerConfig(VeniceProperties serverProperties, Map<String, Map<Str
     kafkaOpenSSLEnabled = serverProperties.getBoolean(SERVER_ENABLE_KAFKA_OPENSSL, false);
     helixHybridStoreQuotaEnabled = serverProperties.getBoolean(HELIX_HYBRID_STORE_QUOTA_ENABLED, false);
     ssdHealthCheckShutdownTimeMs = serverProperties.getLong(SERVER_SHUTDOWN_DISK_UNHEALTHY_TIME_MS, 200000);
+    sslHandshakeThreadPoolSize = serverProperties.getInt(SERVER_SSL_HANDSHAKE_THREAD_POOL_SIZE, 100);
```

Comment:
Does ssl init happen for every query? We are moving from 1 thread to multiple threads which is great, but as you pointed out letting the max possible thread pool size here might be giving too much resources for the ssl init.

Reply:
I don't know how accurate is `We are moving from 1 thread to multiple threads which is great` since the old behavior it will be doing the ssl handshake via caller thread which could spawn a lot more than 100. The idea of using a thread pool executor is actually to be able to limit/control the concurrency of ssl handshakes in case there's a spike of handshake events regardless of the trigger.